graphite/graphite.go

```diff
@@ -43,8 +44,8 @@ func (r *Request) Query(host string) (Response, error) {
 		v.Add("until", fmt.Sprint(r.End.Unix()))
 	}
 	u := url.URL{
-		Scheme:   "http",
-		Host:     host,
+    Scheme:   strings.Split(host, ":")[0],
```

Comment:
url.Parse is not that smart. 
The possible cases we have are:
1. graphiteHost = hostname
2. graphiteHost = hostname:port
3. graphiteHost = protocol:hostname:port
url.Parse gets confused in the various cases so a control flow needs to be put in place.
I'm pushing now a code change that takes care of all cases. 
MM